Sources/CardPayments/CardClient.swift

```diff
@@ -127,11 +129,28 @@ public class CardClient: NSObject {
                     }
                 }
                 
-                let cardResult = CardResult(orderID: orderId, deepLinkURL: url)
-                self.notifySuccess(for: cardResult)
+                guard let url = url else {
+                    self.notifyFailure(with: CardClientError.missingDeeplinkgURLError)
+                    return
+                }
+                
+                if url.absoluteString.contains("state=undefined") && url.absoluteString.contains("code=undefined") {
+                    let liabilityShift = self.getQueryStringParameter(url: url.absoluteString, param: "liability_shift")
+                    let cardResult = CardResult(orderID: orderId, deepLinkURL: url, liabilityShift: liabilityShift)
+                    self.notifySuccess(for: cardResult)
+                } else if url.absoluteString.contains("error=error") {
```

Comment:
Would this be more flexibly if it checked for just that the error key was present, and not that it also equaled to `error`? In case helios made some change to return us proper result details, our SDK wouldn't break.

Reply:
ah that makes sense. so if they in the future had "error=someError1"?